mailet/base/src/main/java/org/apache/mailet/base/MailetUtil.java

```diff
@@ -21,19 +21,107 @@
 
+import com.github.fge.lambdas.Throwing;
+import com.github.fge.lambdas.functions.FunctionChainer;
 import com.google.common.base.Strings;
 
 
 /**
  * Collects utility methods.
  */
 public class MailetUtil {
-    
+
+    public static class ValidationPolicy {
+        public static final ValidationPolicy ALL = new ValidationPolicy(i -> true, "Not expected message");
+        public static final ValidationPolicy STRICTLY_POSITIVE = new ValidationPolicy(i -> i > 0,
+            "Expecting condition to be a strictly positive integer.");
+        public static final ValidationPolicy POSITIVE = new ValidationPolicy(i -> i >= 0,
+            "Expecting condition to be a positive integer.");
+        public static final ValidationPolicy VALID_PORT = new ValidationPolicy(Port::isValid,
+            "Expecting condition to be a valid port.");
+
+        private final Predicate<Integer> isValidPredicate;
+        private final String errorPrefix;
+
+        public ValidationPolicy(Predicate<Integer> isValidPredicate, String errorPrefix) {
+            this.isValidPredicate = isValidPredicate;
+            this.errorPrefix = errorPrefix;
+        }
+
+        public int validate (int value) throws MessagingException {
+            if (!isValidPredicate.test(value)) {
+                throw new MessagingException(errorPrefix + " Got " + value);
+            }
+            return value;
+        }
+
+        public Optional<Integer> validate (Optional<Integer> value) throws MessagingException {
+            FunctionChainer<Integer, Integer> function = Throwing.function(this::validate);
+            return value.map(function.sneakyThrow());
+        }
+    }
+
+    public static class IntegerConditionParser {
+        private Optional<Integer> defaultValue;
+        private Optional<ValidationPolicy> validationPolicy;
+
+        public IntegerConditionParser() {
+            this.defaultValue = Optional.empty();
+            this.validationPolicy = Optional.empty();
+        }
+
+        public IntegerConditionParser withDefaultValue(int defaultValue) {
+            return withDefaultValue(Optional.of(defaultValue));
+        }
+
+        public IntegerConditionParser withDefaultValue(Optional<Integer> defaultValue) {
+            this.defaultValue = defaultValue;
+            return this;
+        }
+
+        public IntegerConditionParser withValidationPolicy(ValidationPolicy validationPolicy) {
+            this.validationPolicy = Optional.of(validationPolicy);
+            return this;
+        }
+
+        public int parse(String condition) throws MessagingException {
+            return parseAsOptional(condition)
+                .orElseThrow(() -> new MessagingException("Condition is required. It should be a strictly positive integer"));
+        }
+
+        public Optional<Integer> parseAsOptional(String condition) throws MessagingException {
+            Optional<String> value = OptionalUtils.or(Optional.ofNullable(condition)
```

Comment:
so you have an integer default value, convert it to string then parse it
what about : Optional.of(condition).map(tryParseInteger).orElse(defaultValue) ?